File: GFG/Filling Bucket/Filling Bucket.cpp
class Solution {
  public:
    int fillingBucket(int N) {
        // code here
        
         if(N<4) return N;
         
        long long mod=1e8;
        vector<long long> dp(N+1);
        dp[0]=0;
        dp[1]=1;
        dp[2]=2;
        for(int i=3;i<=N;i++)
            dp[i]=(dp[i-1]+dp[i-2])%mod;
        return dp[N];
    }
};

File: GFG/Filling Bucket/Filling Bucket.java
class Solution {
    static int fillingBucket(int N) {
        // code here
        if(N==1)return 1;
        int prev=2;
        int secPrev=1;
        int mod=100000000;
        for(int i=0;i<N-2;i++){
            int currAns=(prev%mod+secPrev%mod)%mod;
            secPrev=prev;
            prev=currAns;
        }
        return prev;
    }
};
